src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -617,17 +616,27 @@ class DateBucketPartitioner(gran: Granularity, intervals: Iterable[Interval]) ex
   }
 }
 
-class DateBucketAndHashPartitioner(gran: Granularity, partMap: Map[(Long, Long), Int], optionalDims : Option[Set[String]] = None)
+class DateBucketAndHashPartitioner(gran: Granularity,
+                                   partMap: Map[(Long, Long), Int],
+                                   optionalDims: Option[Set[String]] = None,
+                                   optionalDimExclusions: Option[Set[String]] = None
+                                  )
   extends Partitioner {
-  val maxTimePerBucket = partMap.groupBy(_._1._1).map(e => e._1 -> e._2.size)
-  val dims = optionalDims.orNull
-
+  lazy val shardLookups = partMap.groupBy { case ((bucket, _), _) => bucket }
+    .map { case (bucket, indices) => bucket -> indices.size }
+    .mapValues(maxPartitions => {
+      val shardSpecs = (0 until maxPartitions)
+        .map(new HashBasedNumberedShardSpec(_, maxPartitions, SerializedJsonStatic.mapper).asInstanceOf[ShardSpec])
+        .toList
```

Comment:
force of habit at this point, so many times I ended up doing asJava and ending up with an iterable instead of a list.